File: 백준/Silver/25206. 너의 평점은/너의 평점은.cc
#include <iostream>
using namespace std;

int main() {
	string subject, grade;
	double score;
	double sum = 0, score_sum = 0;

	for (int i = 0; i < 20; i++) {
		cin >> subject >> score >> grade;
		if (grade == "A+") {
			sum += score * 4.5;
			score_sum += score;
		}
		else if (grade == "A0") {
			sum += score * 4.0;
			score_sum += score;
		}
		else if (grade == "B+") {
			sum += score * 3.5;
			score_sum += score;
		}
		else if (grade == "B0") {
			sum += score * 3.0;
			score_sum += score;
		}
		else if (grade == "C+") {
			sum += score * 2.5;
			score_sum += score;
		}
		else if (grade == "C0") {
			sum += score * 2.0;
			score_sum += score;
		}
		else if (grade == "D+") {
			sum += score * 1.5;
			score_sum += score;
		}
		else if (grade == "D0") {
			sum += score * 1.0;
			score_sum += score;
		}
		else if (grade == "F") {
			sum += score * 0.0;
			score_sum += score;
		}
	}

	cout << sum / score_sum;
	return 0;
}

File: 백준/Silver/25206. 너의 평점은/너의 평점은.py
sum = 0
hap = 0
for _ in range(20):
    study, score, grade = input().split()
    score = float(score)
    if grade == 'A+':
        sum += 4.5 * score
    elif grade == "A0":
        sum += 4.0 * score
    elif grade == "B+":
        sum += 3.5 * score
    elif grade == "B0":
        sum += 3.0 * score
    elif grade == "C+":
        sum += 2.5 * score
    elif grade == "C0":
        sum += 2.0 * score
    elif grade == "D+":
        sum += 1.5 * score
    elif grade == "D0":
        sum += 1.0 * score
    elif grade == "F":
        sum += 0.0 * score
    else:
        continue
    hap += score

print(sum / hap)